File: src/org/BennerIndustries/BennerBot/api/datatypes/MessageReference.java
package org.BennerIndustries.BennerBot.api.datatypes;

public class MessageReference {
	String message, source = "";
	UserReference user;
	
	public MessageReference(String message, UserReference user, String source){
		this.message = message;
		this.user = user;
		this.source = source;
	}
	public MessageReference(String message, UserReference user){
		this.message = message;
		this.user = user;
	}
	public String getMessage(){
		return message;
	}
	public UserReference getUser(){
		return user;
	}
	public String getSourcePlugin(){
		return source;
	}
}

File: ConsoleApplication2/MessageReference.cpp
#include "stdafx.h"
#include "MessageReference.h"

using namespace std;

MessageReference::MessageReference(string message, UserReference user, string source) {
	this->message = message;
	this->user = user;
	this->source = source;
}

MessageReference::MessageReference(string message, UserReference user) {
	this->message = message;
	this->user = user;
}

string MessageReference::getMessage() {
	return message;
}

UserReference MessageReference::getUser() {
	return user;
}

string MessageReference::getSourcePlugin() {
	return source;
}